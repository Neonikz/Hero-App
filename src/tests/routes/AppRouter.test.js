import React from 'react'
import { AppRouter } from "../../routers/AppRouter"
import {mount} from 'enzyme';
import { AuthContext } from '../../auth/AuthContext';


describe('Pruebas en <AppRouter />', () => {
    
    const contextValue = {
        dispatch: jest.fn(),
        user:{
            logged:false
        }

    }
    
    test('Debe de mostrar el login si no esta autentiado', () => {
        
        const wrapper = mount( 
            <AuthContext.Provider value={contextValue}>
                <AppRouter /> 
            </AuthContext.Provider>

        );

        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar el componente marvel si esta autenticado', () => {
        
        const contextValue = {
            dispatch: jest.fn(),
            user:{
                logged:true,
                user:'Nicolas'
            }
    
        }

        const wrapper = mount( 
            <AuthContext.Provider value={contextValue}>
                <AppRouter /> 
            </AuthContext.Provider>

        );

        expect( wrapper.find('.navbar').exists() ).toBe(true);
    });
    
    
});
