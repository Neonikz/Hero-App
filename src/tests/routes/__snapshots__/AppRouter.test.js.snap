// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> Debe de mostrar el login si no esta autentiado 1`] = `
<div>
  <div
    className="container text-center mt-5"
  >
    <h1>
      Click on the button to enter
    </h1>
    <hr />
    <button
      className="btn btn-outline-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
